File: app/src/main/java/com/example/development/sakaiclientandroid/fragments/BaseFragment.java
package com.example.development.sakaiclientandroid.fragments;

import android.content.Context;
import android.support.v4.app.Fragment;

public abstract class BaseFragment extends Fragment {

    protected Context mContext;

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        this.mContext = getContext();

    }
}

File: app/src/main/java/com/sakaimobile/development/sakaiclient20/ui/fragments/BaseFragment.kt
package com.sakaimobile.development.sakaiclient20.ui.fragments

import androidx.lifecycle.Observer
import androidx.fragment.app.Fragment
import android.widget.Toast

import com.sakaimobile.development.sakaiclient20.ui.viewmodels.BaseViewModel
import com.sakaimobile.development.sakaiclient20.ui.viewmodels.SakaiErrorState

abstract class BaseFragment : Fragment() {
    fun initRefreshFailureListener(viewModel: BaseViewModel, onRefreshFailed: () -> Unit) {
        viewModel.errorState.observe(viewLifecycleOwner, Observer { error ->
            val message = when(error) {
                SakaiErrorState.SESSION_EXPIRED -> "Your Sakai session has expired, please restart the app."
                SakaiErrorState.FAILURE, null -> "We had trouble reaching Sakai!"
            }

            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
            onRefreshFailed()
        })
    }
}
